### ISSUE TITLE
Place cursor in indented location with automatic brace completion turned on

### ISSUE THREAD

**Issue by cartermp** [Microsoft/visualfsharp#5106]
Today, with automatic brace completion, the cursor will put you where you perhaps don't want it to be:

![image](https://user-images.githubusercontent.com/6309070/41008489-136fd0e2-68e0-11e8-80e2-4d48c652c477.png)

![uhh](https://user-images.githubusercontent.com/6309070/41008643-ec8d12f4-68e0-11e8-8459-9bcc48760db8.gif)

Upon enter, the closing brace is undented appropriately. But the cursor is in a place where I need to reposition it to write code if I want that undentation.

**Comment by cartermp** [comment]
More specifically, upon hitting **Enter**, we should have the matched brace at the end placed on _another_ new line.

```fsharp
let beef = [
    <|> // Cursor here
]
```

**Comment by cartermp** [comment]
This is evidently a regression from VS 2013: https://blogs.msdn.microsoft.com/fsharpteam/2013/06/27/announcing-a-pre-release-of-f-3-1-and-the-visual-f-tools-in-visual-studio-2013/

**Comment by TIHan** [comment]
Yup, based on that post, it is indeed a regression, though not a serious one.
